### PR TITLE
Feature/split caret into two component

### PR DESCRIPTION
**Pull request by nielslyngsoe**
## Description

Replacement of uui-caret into use-case specific components: uui-expand-symbol & uui-sort-symbol

As a result of this uui-caret should be removed completely. (This is though not part of this PR currently)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

- [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
- [ ] I have read the **[CONTRIBUTING](<(https://github.com/umbraco/Umbraco.UI/blob/dev/docs/CONTRIBUTING.md)>)** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/uui-sort-symbol/lib/uui-sort-symbol.element.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    `,
+  ];
+
+  /**
+   * Turns the arrow around.
+   * @type {boolean}
+   * @default false
+   */
+  @property({ type: Boolean, reflect: true })
+  public active = false;
```
Comment (madsrasmussen) [inline packages/uui-sort-symbol/lib/uui-sort-symbol.element.ts:83]: We both use the ActiveMixin and have an active property. Is this needed?

File: packages/uui-sort-symbol/lib/uui-sort-symbol.story.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        color: inherit;
+        border: none;
+        border-bottom: 1px solid currentColor;
+        cursor: pointer;
+      }
+      uui-sort-symbol {
+        margin-left: 20px;
+      }
+      button:hover {
+        --uui-sort-symbol--hover: 1;
```
Comment (madsrasmussen) [inline packages/uui-sort-symbol/lib/uui-sort-symbol.story.ts:42]: You use double dashed before hover in the custom property name. I think this is the only place we do that? I think we should remove that for consistency.

File: packages/uui-sort-symbol/lib/uui-sort-symbol.element.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        opacity: 0;
+        margin-top: -0.5em;
+      }
+
+      #up {
+        transform: rotate(180deg);
+      }
+      #up {
+        margin-top: -0.7em;
+      }
```
Comment (madsrasmussen) [inline packages/uui-sort-symbol/lib/uui-sort-symbol.element.ts:40]: We can combine these

File: packages/uui-sort-symbol/lib/uui-sort-symbol.element.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      #down {
+        margin-top: -0.3em;
+      }
+      :host([active]) #up {
+        margin-top: calc(-0.5em - (0.2em * var(--uui-sort-symbol--hover, 0)));
+      }
+      :host([active]) #down {
+        margin-top: calc(-0.5em + (0.2em * var(--uui-sort-symbol--hover, 0)));
+      }
```
Comment (madsrasmussen) [inline packages/uui-sort-symbol/lib/uui-sort-symbol.element.ts:49]: We can combine these with a comma

File: packages/uui-sort-symbol/lib/uui-sort-symbol.element.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+      :host(:hover) {
+        --uui-sort-symbol--hover: 1;
+      }
+      :host(:not([active])) #up {
+        opacity: calc(0.25 * var(--uui-sort-symbol--hover, 0));
+      }
+      :host(:not([active])) #down {
+        opacity: calc(0.25 * var(--uui-sort-symbol--hover, 0));
+      }
```
Comment (madsrasmussen) [inline packages/uui-sort-symbol/lib/uui-sort-symbol.element.ts:59]: We can combine these with a comma

File: packages/uui-sort-symbol/lib/uui-sort-symbol.element.ts (hunk 1 above)
Comment (nielslyngsoe) [reply packages/uui-sort-symbol/lib/uui-sort-symbol.element.ts:83]: Oh yes, thats a mistake. I have removed that now

File: packages/uui-sort-symbol/lib/uui-sort-symbol.story.ts (hunk 2 above)
Comment (nielslyngsoe) [reply packages/uui-sort-symbol/lib/uui-sort-symbol.story.ts:42]: Good catch, correcting it now :-)

File: packages/uui-sort-symbol/lib/uui-sort-symbol.element.ts (hunk 5 above)
Comment (nielslyngsoe) [reply packages/uui-sort-symbol/lib/uui-sort-symbol.element.ts:59]: oh yes, the result of working forward and back with different approaches. Good one.

**Comment by nielslyngsoe** [comment]
@madsrasmussen thanks for the Review, those things are corrected now :-)

**Pull request merged**